internal/backend/db/dbgorm/builds.go

```diff
@@ -84,11 +80,8 @@ func (db *gormdb) GetBuildData(ctx context.Context, buildID sdktypes.BuildID) ([
 }
 
 func (db *gormdb) ListBuilds(ctx context.Context, filter sdkservices.ListBuildsFilter) ([]sdktypes.Build, error) {
-	builds, err := db.listBuilds(ctx, filter)
-	if builds == nil || err != nil {
-		return nil, translateError(err)
-	}
-
+	bs, err := db.listBuilds(ctx, filter)
+	builds, err := schemasToSDK(bs, err, scheme.ParseBuild)
```

Comment:
I think the a bit more verbose but direct approach that we currently have is better, it's more readable and very simple
schemasToSDK hides error handling that is irrelevant to the conversion from schema to sdk

Reply:
IMO the opposite. It's not hiding the error handling, but clarifies the function and ensures that errors and results are handled in uniform way. The same we(or actually, I) did in `toConnectError`, for example - where errors and messages were handled non-uniformly resulting in different user errors produced by the same flow but for the different objects.
So for me it's not about verbosity, but about uniformity and clarity.

Reply:
I agree that passing error to a function, other than logging or wrapping it feels "off".
The version that checks for errors is more verbose but clearer IMO:

```go
func (db *gormdb) ListBuilds(ctx context.Context, filter sdkservices.ListBuildsFilter) ([]sdktypes.Build, error) {
	bs, err := db.listBuilds(ctx, filter)
	if err != nil {
		return nil, err
	}

	builds, err := schemasToSDK(bs, scheme.ParseBuild)
	if err != nil {
		return nil, err
	}

	slices.Reverse(builds)
	return builds, nil
}
```

This quotes comes to mind:
![GSiQThWXwAAnz_v](https://github.com/user-attachments/assets/0ca2c39e-c936-4a1d-858a-3a800a9c8932)